cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -125,6 +125,14 @@ struct arrow_schema_data_types {
   data_type type{type_id::EMPTY};
 };
 
+/**
+ * @brief Struct to store the number of row groups surviving each predicate pushdown filter.
+ */
+struct num_surviving_row_groups {
```

Comment:
Nit:  The fact that this is a struct, but named `num_surviving...` feels off.  But calling it `row_group_filter_survival_stats` overloads the `stats`.

`row_group_filter_survival_metrics` might work.  Again, this is a trivial nit, and not worth holding this up for.